doc/source/conf.py

```diff
@@ -368,3 +368,19 @@
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 #
 # texinfo_no_detailmenu = False
+
+
+def setup(app):
+    """ Entry point to sphinx build customisation. """
+    app.connect('autodoc-skip-member', special_methods_callback)
+
+
+def special_methods_callback(app, what, name, obj, skip, options):
+    """
+    Include all method and function docstrings in the docs if they have a custom docstring. This
+    overrides the default behaviour of omitting special methods or functions from the docs.
+    """
+    if hasattr(obj, '__doc__') and isinstance(obj, (types.FunctionType, types.MethodType)):
+        return False
```

Comment:
```suggestion
        return False  # never skip when a function/method has a docstring
```

Reply:
i'm averse to comments - do you not feel the docstring covers it?

Reply:
I added the comment because it took me a bit to understand what exactly is going on here - I could figure it out by myself, but wanted to save future-me and other people having to go through that extra effort

Reply:
i've edited the docstring - is that clearer?